tests/suites/0_stateless/05_hints/05_0001_set_var.sh

```diff
@@ -28,4 +28,9 @@ echo "select * from set_var.test order by id" | $MYSQL_CLIENT_CONNECT
 echo "delete /*+SET_VAR(timezone='Asia/Shanghai') SET_VAR(storage_read_buffer_size=200)*/ from set_var.test where id=2" | $MYSQL_CLIENT_CONNECT
 echo "delete /*+SET_VAR(timezone='Asia/Shanghai') (storage_read_buffer_size=200)*/ from set_var.test where id=4" | $MYSQL_CLIENT_CONNECT
 echo "select * from set_var.test" | $MYSQL_CLIENT_CONNECT
+
+echo "set timezone='America/Toronto'; select /*+SET_VAR(timezone='Asia/Shanghai') */ timezone(); select timezone();" | $MYSQL_CLIENT_CONNECT
+echo "create table set_var.t(c1 timestamp)" | $MYSQL_CLIENT_CONNECT
+# Toronto and Shanghai time diff is 13 hours.
+echo "set timezone='America/Toronto'; insert /*+SET_VAR(timezone='Asia/Shanghai') */ into set_var.t values('2022-02-02 03:00:00'); select /*+SET_VAR(timezone='Asia/Shanghai') */ * from set_var.t; select * from set_var.t;" | $MYSQL_CLIENT_CONNECT
```

Comment:
what happened if
```
"SET timezone='America/Toronto' SELECT /*+SET_VAR(timezone='Asia/Shanghai')*/ * FROM system.settings where name = 'timezone' and value in (select value from system.settings where name = 'timezone')"
```

Reply:
Will return empty.

```
MySQL [(none)]> set timezone='America/Toronto'; SELECT /*+SET_VAR(timezone='Asia/Shanghai')*/ * FROM system.settings where name = 'timezone' and value in (select value from system.settings where name = 'timezone');
Query OK, 0 rows affected (0.007 sec)

Empty set (0.027 sec)
```

I think that's is ok? In MySQL doc:


> The [SET_VAR](https://dev.mysql.com/doc/refman/8.0/en/optimizer-hints.html#optimizer-hints-set-var) hint is permitted only at the statement level. If used in a subquery, the hint is ignored with a warning.

Reply:
> what happened if
> 
> ```
> "SET timezone='America/Toronto' SELECT /*+SET_VAR(timezone='Asia/Shanghai')*/ * FROM system.settings where name = 'timezone' and value in (select value from system.settings where name = 'timezone')"
> ```

It's ok I think.